doc/lua_api.txt

```diff
@@ -3261,6 +3261,13 @@ Player Inventory lists
 * `hand`: list containing an override for the empty hand
     * Is not created automatically, use `InvRef:set_size`
     * Is only used to enhance the empty hand's tool capabilities
+* `offhand`: list containing the offhand wielded item.
+    * Will be used for placements and secondary uses if the
+        main hand does not have any node_place_prediction, on_place
+        or on_secondary_use callbacks.
+    * Is passed to on_place and on_secondary_use callbacks; make sure
+        mods are aware of the itemstack not neccessarily being
+        located in the main hand.
```

Comment:
The offhand can only be used for RMB action, punches are always executed with the main hand, so for `on_punch`, this is expected behavior.

And `on_rightclick` is affected by this PR, I have tested it.

Reply:
> How can you make this sure in existing mods?

Have a look at the source code, make sure they dont call set_wielded_item based on assumptions.

This is essentially a 'heads up' for game devs / modders / server admins. If they add dual wielding code, they have to make sure that all mods they use are actually compatible, if necessary patching them. Otherwise, things may break.

This is not optimal, but dual wielding is disabled by default for a reason. New features should not be prevented by the possibility of breakage, as long as it's possible to disable the feature.